update-repo.py

```diff
@@ -0,0 +1,166 @@
+        lines = file.readlines()
+
+    # Look for the line containing 'project:' and update it
+    updated_lines = []
+    for line in lines:
+        if 'project:' in line:
+            updated_lines.append(f'      project: {package_name}\n')
+        else:
+            updated_lines.append(line)
+
+    # Write the updated lines back to the file
+    with open(file_path, 'w') as file:
+        file.writelines(updated_lines)
+
+
+def copy_file(source_file, dirpath, workflow_dir, dest_file, username):
+    # Create and switch to a new branch named after the new version
+    run_command(f"git checkout -b {source_file}", cwd=dirpath)
+
+    # Copy the file
+    shutil.copy2(source_file, dest_file)
+    print(f'File copied to {workflow_dir}')
+
+    # Pass the package name into the file
+    package_name = dirpath.split('/')[1]
+    pass_variable(dest_file, package_name)
+
+    # Create PR
+    if package_name.split('.')[0] == "diffpy":
+        org_name = "diffpy"
+    else:
+        org_name = "Billingegroup"
+    create_PR(dirpath, source_file, username, package_name, org_name)
+
+
+def create_PR(cwd, file, username, package_name, org_name):
+    """
+    Create a PR from a branch name of <new_version>
+    to the main branch of the feedstock repository.
+    """
+
+    run_command(f"git add workflows/{file}", cwd=cwd)
+
+    # Commit the changes
+    run_command(f'git commit -m "Add {file} to workflow"', cwd=cwd)
+
+    # Push the new branch to your origin repository
+    run_command(f"git push origin {file}", cwd=cwd)
+
+    run_command(f"gh repo set-default {org_name}/{package_name}", cwd=cwd)
+
+    # Create a pull request using GitHub CLI
+    pr_command = (
+        f"gh pr create --base main --head {username}:{file} "
+        f"--title 'Add {file} to workflows' "
+        f"--body 'Added {file} to workflows'"
+    )
+
+    # Run the PR create command in the appropriate directory
+    run_command(pr_command, cwd=cwd)
+
+
+"""
+GitHub Integration
+"""
+
+
+def get_github_username():
+    """Get the GitHub username using the GitHub CLI."""
+    try:
+        username = subprocess.check_output(
+            ["gh", "api", "user", "--jq", ".login"], text=True
+        ).strip()
+        return username
+    except subprocess.CalledProcessError:
+        raise RuntimeError(
+            "Could not retrieve GitHub username using GitHub CLI. "
+            "Please make sure your local machine is authenticated with GitHub."
+        )
+
+
+"""
+Main Entry Point
+"""
+
+
+def main():
+    source_file_path = sys.argv[1]
+    repo_file_path = sys.argv[2]
+
+    source_file = source_file_path.split("/")[-1]
+    dest_dir = repo_file_path.split("/")[-1]
+
+    # Get the GitHub username using the GitHub CLI
+    username = get_github_username()
+
+    for dirpath, dirnames, filenames in os.walk('./'):
+        if dest_dir in dirnames:
+            workflow_dir = os.path.join(dirpath, str(dest_dir))  # Full path to the workflow directory
```

Comment:
please set variables/constants that do not change during for-loop above the loop

Reply:
workflow_dir changes because the walk is performed on the root directory, so workflow_dir would change based on which repo I'm currently in. 2 examples would be diffpy.utils/.github/workflows and diffpy.snmf/.github/workflows